cmd/jcsd/main.go

```
@@ -0,0 +1,79 @@
+	encoder := json.NewEncoder(conn)
+
+	var req Request
+	if err := decoder.Decode(&req); err != nil {
+		fmt.Fprintln(os.Stderr, "Decode error:", err)
+		encoder.Encode(Response{Error: "invalid JSON"})
+		return
+	}
+
+	if req.Op != "canon" {
+		fmt.Fprintln(os.Stderr, "Unsupported op:", req.Op)
+		encoder.Encode(Response{Error: "unsupported op"})
```

golangci-lint (1.64.8)

Error return value of `encoder.Encode` is not checked

(errcheck)